airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -70,14 +70,48 @@ definitions:
         type: string
   AirbyteStateMessage:
     type: object
-    additionalProperties: true
-    required:
-      - data
+    additionalProperties: false
     properties:
+      type:
+        "$ref": "#/definitions/AirbyteStateType"
       data:
-        description: "the state data"
-        type: object
-        existingJavaType: com.fasterxml.jackson.databind.JsonNode
+        description: "Deprecated"
+        "$ref": "#/definitions/AirbyteStateBlob"
+      global:
+        "$ref": "#/definitions/AirbyteStateBlob"
+      streams:
+        type: array
+        items:
+          "$ref": "#/definitions/AirbyteStreamState"
+
+  AirbyteStateType:
+    type: string
+    description: >
+      The type of state the other fields represent.
+      If not set, the state data is interpreted as GLOBAL and should be read from the `data` field for backwards compatibility.
+      GLOBAL means that the state should be read from `global` and means that it represents the state for all the streams.
+      PER_STREAM means that the state should be read from `streams`. Each item in the ist represents the state for the associated stream.
+    enum:
+      - GLOBAL
+      - PER_STREAM
+
+  AirbyteStreamState:
+    type: object
+    description: "per stream state data"
+    additionalProperties: false
+    properties:
+      name:
```

Comment:
```suggestion
      stream_name:
```

Reply:
isn't it redundant with the name of the fields in the `AirbyteSateMessage` it will be something like stream[x].stream_name.

Reply:
I don't feel strongly on this one, but I think I do lean towards sherif's suggestion. I don't think it's redundant because when you say an `AirbyteStreamState` has a `name` it isn't necessarily obvious that that name is a stream name. It could be interpreted as `AirbyteStreamState`s just have a name for some reason.  doing with `stream_name` removes the ambiguity.